DataStructures/ArrayBasedStack/ArrayBasedStack.cs

```diff
@@ -70,30 +75,45 @@ public int Capacity
         /// <summary>
         /// Removes all items from the <see cref="ArrayBasedStack{T}"/>.
         /// </summary>
-        public void Clear() => count = 0;
+        public void Clear()
+        {
+            top = -1;
+            Capacity = DefaultCapacity;
+        }
 
         /// <summary>
         /// Determines whether an element is in the <see cref="ArrayBasedStack{T}"/>.
         /// </summary>
         /// <param name="item">The item to locate in the <see cref="ArrayBasedStack{T}"/>.</param>
         /// <returns>True, if the item is in the stack.</returns>
-        public bool Contains(T item) => Array.IndexOf(stack, item, 0, count) > -1;
+        public bool Contains(T item) => Array.IndexOf(stack, item, 0, top + 1) > -1;
 
         /// <summary>
         /// Returns the item at the top of the <see cref="ArrayBasedStack{T}"/> without removing it.
         /// </summary>
         /// <returns>The item at the top of the <see cref="ArrayBasedStack{T}"/>.</returns>
-        public T Peek() => stack[count - 1];
+        public T Peek()
+        {
+            if (top == -1)
+            {
+                throw new ApplicationException(StackEmptyErrorMessage);
```

Comment:
![image](https://user-images.githubusercontent.com/29201949/108167219-06061c80-70fe-11eb-8a8b-549895f9053f.png)

Reply:
Try switching to InvalidOperationException

Reply:
Thanks for the review. Let me change this and commit.